src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            machines.addAll(new BulldozerProducer().get());
+            return machines;
+        } else if (type == Truck.class) {
+            machines.addAll(new TruckProducer().get());
+            return machines;
+        } else if (type == Excavator.class) {
+            machines.addAll(new ExcavatorProducer().get());
+            return machines;
+        }
+        return machines;
```

Comment:
Keep if-else constructions simple. Since each if condition returns a value, there is no need for an else block. You could remove the else keyword to simplify the code flow.